app/models/spree/order_contents_decorator.rb

```diff
@@ -1,5 +1,26 @@
 module Spree
   OrderContents.class_eval do
+
+    def add_to_line_item(variant, quantity, options = {})
+      line_item = grab_line_item_by_variant(variant, false, options)
+
+      if line_item && part_variants_match?(line_item, variant, quantity, options)
+        line_item.quantity += quantity.to_i
+        line_item.currency = currency unless currency.nil?
+      else
+        opts = { currency: order.currency }.merge ActionController::Parameters.new(options).
```

Comment:
Why do we go through `ActionController::Parameters.new` here?
Would it work with `{ currency: order.currency }.merge(options)`?